packages/framework/src/client/scheduler.js

```diff
@@ -0,0 +1,140 @@
+import { find } from 'lodash';
+import ResourceRequest, { STATUS } from './resource-request';
+
+const DEFAULT_FETCH_INTERVAL = 5000; // Five seconds
+
+export default class Scheduler {
+	constructor(
+		fetch,
+		fetchInterval = DEFAULT_FETCH_INTERVAL,
+		setInterval = window.setInterval,
+	) {
+		this.fetch = fetch;
+		this.requests = [];
+		this.timerId = null;
+
+		if ( fetchInterval > 0 ) {
+			setInterval( () => {
```

Comment:
This interval should be stored somewhere so it can be stopped (for example, to handle when the page is not visible).

Also, why are you using `setInterval` instead of the more common pattern of using `setTimeout` and creating a new `setTimeout` at the end of the callback?

Reply:
> This interval should be stored somewhere so it can be stopped (for example, to handle when the page is not visible).

You're completely right. I added `clearInterval` to #202 for this very reasons. If you'd like me to move that bit of code to this PR, let me know.

> Also, why are you using setInterval instead of the more common pattern of using setTimeout and creating a new setTimeout at the end of the callback?

Well, in this case I see the scheduler running at the same interval for the lifetime of the application, so `setInterval` seemed like the right tool for the job. I haven't used `setInterval` much before though. Are there any pitfalls associated with it that a repeating `setTimeout` wouldn't have? The only one I can think of is what happens if a single run of the interval takes longer than the time interval set. I don't think it's possible to run the interval twice at the same time though, right?

Reply:
> If you'd like me to move that bit of code to this PR, let me know.

No, that's fine :)

> Are there any pitfalls associated with it that a repeating setTimeout wouldn't have?

It's just that `setTimeout` gives you more control with the timing, because you set the next "tick" explicitly. As you said, there's an issue if the function takes too long to run: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval#Ensure_that_execution_duration_is_shorter_than_interval_frequency

But I don't think this is important here. Maybe `setTimeout` will be better to easily adjust the interval if the tab is inactive, for example.

Reply:
Yeah, that's a good point. I'll change it to a recurring `setTimeout` just in case execution takes longer (especially because I speed up the interval in the Step 3 PR). And you're right about the tab being inactive too.

Reply:
I've updated this PR to use `setTimeout` instead and removed the interval altogether. It now calculates the next delay every time a request is scheduled and after each processing run.